### PR TITLE
configurations as used in alignment with first data (CRUZET+CRAFT 3.8T)

### PR DESCRIPTION
**Pull request by eavdeeva**
configuration files as they were used in alignment with first 2015 data (CRUZET+CRAFT 3.8T)

### CODE REVIEW

File: Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        )
+    ),
+    destinations = cms.untracked.vstring('alignment') ## (, 'cout')
+
+)
+
+#-- Magnetic field
+#process.load("Configuration.StandardSequences.MagneticField_AutoFromDBCurrent_cff") # for data
+process.load("Configuration/StandardSequences/MagneticField_38T_cff") ## FOR 3.8T
+#process.load("Configuration.StandardSequences.MagneticField_0T_cff")  ## FOR 0T
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py]: Hi @eavdeeva is this still needed?

File: Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+## if alignment constants not from global tag, add this
+from CondCore.DBCommon.CondDBSetup_cfi import *
+#####################################################################
+## Global Tag
+######################################################################
+
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+process.GlobalTag.connect = "frontier://FrontierProd/CMS_COND_31X_GLOBALTAG"
+process.GlobalTag.globaltag = "GR_P_V49::All" #"GR_P_V16::All" 
+################################################################
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py]: please move to condDBv2

File: Alignment/MillePedeAlignmentAlgorithm/test/deadmodules.txt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    # TEC
+    #470340521, 470063045, 470063046, 470114669, 470114670, 470161093, 470161094, 470164333, 470164334, 470312005, 470312006, 470312009, 470067405, 470067406, 470128813
+344070404, #outlier: large rotation around gamma (0.2 rad)
+402672394, 
+352459012,
+344070404,
+369125398,
+369141862,
+470116592,
+
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/deadmodules.txt]: the reference points to a 2008 HN. Are your really still using that bad module list?

File: Alignment/MillePedeAlignmentAlgorithm/test/setup_align.pl [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+my $home = $ENV{'HOME'};
+
+my $mpsdir = "$ENV{'CMSSW_BASE'}/src/Alignment/MillePedeAlignmentAlgorithm/scripts";
+
+my $append = 0;
+
+my $datasetdir = '/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/MP/MPproduction/datasetfiles';
+my $jsondir = '/afs/cern.ch/cms/CAF/CMSALCA/ALCA_TRACKERALIGN/MP/MPproduction/datasetfiles';
+
+my $promptglobaltag = 'GR_P_V49::All';
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/setup_align.pl]: move to condDBv2

File: Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py (hunk 1 above)
Comment (diguida) [inline Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_forcefield38T.py]: @eavdeeva 

````
Configuration/StandardSequences/MagneticField_38T_cff
````

is obsolete. Please use

````
process.load("Configuration.StandardSequences.MagneticField_AutoFromDBCurrent_cff")
````

**Comment by diguida** [comment]
@eavdeeva please make corresponding PR in 75X and 76X.

**Comment by mmusich** [comment]
-1
in the present state of obsolence of the script, the chagnes cannot be merged.

File: Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_0T.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+
+## if alignment constants not from global tag, add this
+from CondCore.DBCommon.CondDBSetup_cfi import *
+#####################################################################
+## Global Tag
+######################################################################
+
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+process.GlobalTag.connect = "frontier://FrontierProd/CMS_COND_31X_GLOBALTAG"
+process.GlobalTag.globaltag = "GR_P_V49::All" #"GR_P_V16::All"
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_0T.py]: please move to condDBv2

File: Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_0T.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+
+#placeholderalignables
+process.AlignmentProducer.checkDbAlignmentValidity = False
+
+#process.AlignmentProducer.doMuon = True # to align muon system
+
+process.AlignmentProducer.doMisalignmentScenario = True
+# If the above is true, you might want to choose the scenario:
+from Alignment.TrackerAlignment.Scenarios_cff import *
+process.AlignmentProducer.MisalignmentScenario = TECRing7Minus133mmScenario
```
Comment (mmusich) [inline Alignment/MillePedeAlignmentAlgorithm/test/alignment_cosmics_0T.py]: @eavdeeva @mschrode can you clarify if forcing the TEC R7 fix is still in use? I thought this was fixed already in cosmics

**Comment by eavdeeva** [comment]
Hi Marco, Salvatore,

these are the configurations exactly as were used to prepare millepede alignment candidate on CRUZET+CRAFT data with CMSSW_7_4_0. Now, with collisions data, we are using different configurations but they are not finalized, and we want to keep configurations as they were used for cosmics alignment while working towards preparing better configurations and better alignment with collision data, and keeping in mind that we would update configs once they are finalized. Please, let us know what strategy would you recommend given that our configs used for cosmics alignment are obsolete but we still want to keep them as reference. 

Thank you!

**Comment by eavdeeva** [comment]
Marco,

TEC Ring 7 was not applied in mp1598 which was the very first alignment with CRUZET data, that's why we applied the correction for the newest alignment with cosmics which was performed on top of mp1598. Current collision alignment use newest cosmics alignment as start geometry and thus TEC Ring 7 is not applied for the alignments we are running now

**Comment by diguida** [comment]
@eavdeeva 
Let me remind you what is one of the purposes of including configuration files in release: reproducibility of results.
Given that the release evolves, you should keep them up to date.
Now, as in the 74X release cycle there have been changes w.r.t. the environment you originally used, you should do the following:

1. run the configuration as it is;
2. edit/commit in case of problems;
3. check that is consistent with the CMSSW environment (i.e. Condition DB changeover)
4. test that the final configuration reproduces exactly the original results.

Hope this clarifies

**Comment by frmeier** [comment]
@diguida 
I agree with your comment in general. But keeping things documented as they were done is also an important step. In part, @eavdeeva did this PR on my request. As she laid out, this configuration is what was used to produce the object a few weeks ago. In the meantime, things evolved and several (mis-)configurations have been resolved and the configuration we would use today would look different, for sure.

The alignment object in question is about to become obsolete with incoming data and will not be reproduced. But we may have to go back to what was used - with all shortcomings this config had from today's perspective. If we don't put this into github, all the documentation trace-back we will likely have are the MP-areas on AFS. In the past we already lost some configs because it is so easy to make an accidental `rm mp*`. But if you can accept the possible loss of documentation, then we can do otherwise.

Just my 2 cents.

**Comment by mmusich** [comment]
@frmeier, I agree with you about the need to document the configuration used to run the first alignment on run2 data, but we cannot allow obsolete configuration to sneak in the release otherwise in a couple of year everybody will have forgotten what was the purpose of these files, and believe they are the state of the art. IMHO a dedicated branch (with a self-explaining name) at https://github.com/cms-trackeralign/cmssw is the best place to push configurations of historical interest. Please help us in keeping the release clean. 
Cheers

Marco

**Pull request closed without merging**